src/controllers/todo.controller.ts

```diff
@@ -0,0 +1,67 @@
+import { post, param, get, put, patch, del } from '@loopback/rest';
+import { TodoSchema, Todo, TodoItem } from '../models';
+import { repository, EntityCrudRepository } from '@loopback/repository';
+
+export class TodoController {
+  constructor(
+    @repository('todo') protected todoRepo: EntityCrudRepository<Todo, number>,
+    @repository('todo-item')
+    protected todoItemRepo: EntityCrudRepository<TodoItem, number>
+  ) {}
+  @post('/todo')
+  @param.body('todo', TodoSchema)
+  async create(todo: Todo) {
+    let checklist = todo.checklist;
+    delete todo.checklist;
+    const result = await this.todoRepo.create(todo);
+    if (checklist) {
+      for (const item in checklist) {
+        checklist[item].todoId = result.id;
+        let newItem = await this.todoItemRepo.create(checklist[item]);
+        checklist[item] = newItem;
+      }
+    }
+    result.checklist = checklist;
+    return result;
+  }
+
+  @get('/todo/{id}')
+  @param.path.number('id')
+  @param.query.boolean('items')
+  async findById(id: number, items?: boolean): Promise<Todo> {
+    const todo = await this.todoRepo.findById(id);
+    if (todo && items) {
+      todo.checklist = await this.todoItemRepo.find({
+        where: {
+          todoId: todo.id
+        }
+      });
+    }
+    return todo;
+  }
+
+  @get('/todo')
+  async find(): Promise<Todo[]> {
+    return await this.todoRepo.find();
+  }
+
+  @put('/todo/{id}')
+  @param.path.number('id')
+  @param.body('todo', TodoSchema)
+  async replace(id: number, todo: Todo): Promise<boolean> {
```

Comment:
IIUC, here `return await this.todoRepo.replaceById(id, todo);` uses the function `replaceById` in `juggler/dao.js`.

If your `todo` instance contains `[todoItems]`, `replaceById` treats it as an embedded item. 

While from your implementation of `create` and `get` method, `todo` and `todoItem` are stored separately, not like one embeds another.

For a memory db, embeds and separately stored both look good, but a mix of the use might cause inconsistency...?

 For example you replace 
`oldToDo: {id: 1, title: 'todoFoo'}` 
with 
`newToDo: {id: 1, title: 'todoBar', todoItem: [{id: 100}, {id: 200}]}`, 
then `GET/todo/1?true` won't return you the todo items.

Reply:
probably comment above is too picky in a "structure review" , I think my point is, without leveraging juggler's relation mixin, it turns to the developer's job to unify the logic that connects two(or more) models across all methods :)

Reply:
would using the [relation decorators](http://loopback.io/doc/en/lb4/Decorators.html#relation-decorators) make sense in this case?

Reply:
I want to avoid use of relation decorators since I'm not certain they work, and they don't need to be a part of the beta push.

Reply:
@b-admike I think relation decorator is not implemented yet, so it doesn't play a validation/restriction role here :(

Reply:
oh okay, makes sense.